contracts/staking/TokenStaking.sol

```diff
@@ -477,22 +335,62 @@ contract TokenStaking is Initializable, IStaking, Checkpoints {
         StakingProviderInfo storage stakingProviderStruct = stakingProviders[
             stakingProvider
         ];
-        AppAuthorization storage authorization = stakingProviderStruct
-            .authorizations[application];
-        uint96 fromAmount = authorization.authorized;
-        require(fromAmount > 0, "Application is not authorized");
-        authorization.authorized = 0;
-        authorization.deauthorizing = 0;
-
-        emit AuthorizationDecreaseApproved(
+        forceDecreaseAuthorization(
             stakingProvider,
-            application,
-            fromAmount,
-            0
+            stakingProviderStruct,
+            application
         );
         cleanAuthorizedApplications(stakingProviderStruct, 1);
     }
 
+    // TODO consider rename
```

Comment:
I think this name is ok. Or at least I can't think of a better one.

Reply:
✔️